src/components/common/TmSessionExtension.vue

```diff
@@ -76,13 +76,31 @@ export default {
       })
     },
     async signInAndRedirect(account) {
+      let accountNetwork = this.networks.find(
+        ({ id }) => id === account.network
+      )
+      // if no network, trying to identify by network_prefix
+      if (!accountNetwork) {
+        // select mainnet first
+        accountNetwork = this.networks.find(
+          ({ address_prefix, testnet }) =>
+            account.address.startsWith(address_prefix) && !testnet
```

Comment:
doesn't work with Polkadot

Reply:
create helper function `getNetworkFromAddress`

Reply:
I thought Polkadot had now an address_prefix, 2

Reply:
but the prefix is not a prefix. it is a value to calculate the address from the private key

Reply:
I see. It is a "fake" prefix

Reply:
This is not the issue with Polkadot. Mainly this change should help users with old extension, having no network field in their accounts. But those cases are numerous.

Reply:
oh sure, and those ones cannot have Polkadot accounts logically

Reply:
But the logic is confusing. Why didn't you use the same logic already existing in `TmSessionEplore`?

Reply:
Cause we have no testnet property in this component and we don't have addressPrefixes too.

Reply:
Yap, you are right...

Reply:
```
      // HACK as polkadot addresses don't have a prefix
      if (isPolkadotAddress(this.address) && this.testnet) {
        return this.networks.find(({ id }) => id === "polkadot-testnet")
      }

      const selectedNetworksArray = this.networks.filter(({ address_prefix }) =>
        this.address.startsWith(address_prefix)
      )
```

Now find the network `selectedNetworksArray` based on the testnet prop.